cli/src/upload.rs

```diff
@@ -0,0 +1,371 @@
+use std::env;
+use std::io::BufReader;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use crate::{
+    api_client::ApiClient,
+    bundler::BundlerUtil,
+    codeowners::CodeOwners,
+    constants::{EXIT_FAILURE, EXIT_SUCCESS},
+    runner::{build_filesets, extract_failed_tests, run_quarantine},
+    scanner::{EnvScanner, FileSet},
+    types::{BundleMeta, QuarantineBulkTestStatus, QuarantineRunResult, META_VERSION},
+    utils::parse_custom_tags,
+};
+use api::BundleUploadStatus;
+use clap::Args;
+use context::junit::parser::JunitParser;
+use context::repo::BundleRepo;
+#[cfg(target_os = "macos")]
+use xcresult::XCResult;
```

Comment:
nit: let's clean up these imports a little, to be consistent. We don't need to use `std::blah::Blah` everywhere below. Also, I think we're missing the import of `std::io::Write` (I added it in my suggestion)
```suggestion
use std::{
    env, fs,
    io::{BufReader, Write},
    path,
    time::{SystemTime, UNIX_EPOCH},
};

use crate::{
    api_client::ApiClient,
    bundler::BundlerUtil,
    codeowners::CodeOwners,
    constants::{EXIT_FAILURE, EXIT_SUCCESS},
    runner::{build_filesets, extract_failed_tests, run_quarantine},
    scanner::{EnvScanner, FileSet},
    types::{BundleMeta, QuarantineBulkTestStatus, QuarantineRunResult, META_VERSION},
    utils::parse_custom_tags,
};
use api::BundleUploadStatus;
use clap::Args;
use context::{junit::parser::JunitParser, repo::BundleRepo};
#[cfg(target_os = "macos")]
use xcresult::XCResult;
```